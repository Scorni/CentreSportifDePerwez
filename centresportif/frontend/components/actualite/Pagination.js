import react from "react";
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import {perPage } from '../../config';
import Head from 'next/head';
import Link from 'next/link';
import { Button } from 'reactstrap';

const PAGINATION_QUERY = gql`
    query PAGINATION_QUERY
        {
        actualitiesConnection{
            aggregate {
                count
            }
        }
    }
`;
const Pagination = props => {
    return(
        <>
        <Query query={PAGINATION_QUERY}>
            {({data,loading,error}) =>{
                if (loading) return <p>Loading...</p>
                    const count = data.actualitiesConnection.aggregate.count;
                    const pages = Math.ceil(count / perPage);
                    const page = props.page;
                return(
                        <>
                        <Head>
                            <title>
                                Actualité Page {page} sur {pages}
                            </title>
                        
                        </Head>
                        <Link 
                            prefetch 
                            href={{
                                pathname: 'actuality',
                                query: { page: page - 1}
                            }}
                        >
                            <Button disabled={page <= 1}><a className="precedent" role="button" >Précédent</a></Button>
                        </Link>
                        <p>Page {page} sur {pages}</p>
                        <p>Nombre d'actualités : {count}</p>
                        <Link 
                            prefetch 
                            href={{
                                pathname: 'actuality',
                                query: { page: page + 1}
                            }}
                        >
                            <Button disabled={page >= pages}><a className="suivant" role="button" >Suivant</a></Button>
                        </Link>
                    </>
            )}}
        </Query>
            
        </>
    )
    
}

export default Pagination;