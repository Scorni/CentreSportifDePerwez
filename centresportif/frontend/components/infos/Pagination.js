import react from "react";
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import {perPage } from '../../config';
import Head from 'next/head';
import Link from 'next/link';
import { Button } from 'reactstrap';

const PAGINATION_QUERY = gql`
    query PAGINATION_QUERY
        {
        faqsConnection{
            aggregate {
                count
            }
        }
    }
`;
const Pagination = props => {
    return(
        <>
        <Query query={PAGINATION_QUERY}>
            {({data,loading,error}) =>{
                if (loading) return <p>Loading...</p>
                    const count = data.faqsConnection.aggregate.count;
                    const pages = Math.ceil(count / perPage);
                    const page = props.page;
                return(
                        <div className="customPaginationDiv" data-test="pagination">
                            <Head>
                                <title>
                                    FAQ Page {page} sur 
                                        {pages}
                                </title>
                            
                            </Head>
                            <Link 
                                 
                                href={{
                                    pathname: 'actuality',
                                    query: { page: page - 1}
                                }}
                            >
                                <Button disabled={page <= 1} className="customActualityButtonPage precedentButton"><a className="precedent" role="button" >Précédent</a></Button>
                            </Link>
                            <Button className="customButtonPagination" disabled>Page {page} sur <span className="totalPages">{pages}</span></Button>
                            <Link 
                                 
                                href={{
                                    pathname: 'faq',
                                    query: { page: page + 1}
                                }}
                            >
                                <Button disabled={page >= pages} className="customActualityButtonPage suivantButton"><a className="suivant" role="button" >Suivant</a></Button>
                            </Link>
                        </div>
            )}}
        </Query>
            
        </>
    )
    
}

export default Pagination;
export {PAGINATION_QUERY}