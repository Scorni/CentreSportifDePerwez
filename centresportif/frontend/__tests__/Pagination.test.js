import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import Pagination, { PAGINATION_QUERY} from '../components/infos/Pagination';
import {MockedProvider} from 'react-apollo/test-utils';

function makeMocksFor(length){
    return [
        {
            request : { query: PAGINATION_QUERY},
            result : {
                data:{
                    faqsConnection: {
                        __typename: 'aggregate',
                        aggregate:{
                            __typename:'count',
                            count: length
                        }
                    }
                }
            }
        }
    ]
}

describe('<Pagination/>', () => {
    it('display things', () => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(1)}>
                <Pagination page={1} />
            </MockedProvider>
        );
        const pagination = wrapper.find('[data-test="pagination"]')
        expect(wrapper.text()).toContain('Loading...')
        //expect(toJSON(pagination)).toMatchSnapshot();
        //console.log(wrapper.debug())
    })
    it('renders pagination for 15 actualities', async () => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(15)}>
                <Pagination page={1} />
            </MockedProvider>
        );
        await wait();
        wrapper.update();
        expect(wrapper.find('.totalPages').text()).toEqual('3');
        const pagination = wrapper.find('[data-test="pagination"]')
        expect(toJSON(pagination)).toMatchSnapshot();
        //console.log(wrapper.debug())
    })
    it('disable previous button on first page',async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(15)}>
                <Pagination page={1} />
            </MockedProvider>
        );
        await wait();
        wrapper.update();
        expect(wrapper.find('Button.precedentButton').prop('disabled')).toEqual(true)
        expect(wrapper.find('Button.suivantButton').prop('disabled')).toEqual(false)

    })
    it('disable next button on last page',async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(15)}>
                <Pagination page={3} />
            </MockedProvider>
        );
        await wait();
        wrapper.update();
        expect(wrapper.find('Button.precedentButton').prop('disabled')).toEqual(false)
        expect(wrapper.find('Button.suivantButton').prop('disabled')).toEqual(true)
    })
    it('enable all buttons on middle page',async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(15)}>
                <Pagination page={2} />
            </MockedProvider>
        );
        await wait();
        wrapper.update();
        expect(wrapper.find('Button.precedentButton').prop('disabled')).toEqual(false)
        expect(wrapper.find('Button.suivantButton').prop('disabled')).toEqual(false)
    })
})