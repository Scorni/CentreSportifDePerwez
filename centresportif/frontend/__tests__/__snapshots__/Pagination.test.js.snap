// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> renders pagination for 15 actualities 1`] = `
<div
  className="customPaginationDiv"
  data-test="pagination"
>
  <Head>
    <_default
      headManager={Object {}}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
  <Link
    href={
      Object {
        "pathname": "actuality",
        "query": Object {
          "page": 0,
        },
      }
    }
  >
    <Button
      className="customActualityButtonPage precedentButton"
      color="secondary"
      disabled={true}
      onClick={[Function]}
      onMouseEnter={[Function]}
      tag="button"
    >
      <button
        aria-label={null}
        className="customActualityButtonPage precedentButton btn btn-secondary disabled"
        disabled={true}
        onClick={[Function]}
        onMouseEnter={[Function]}
        type="button"
      >
        <a
          className="precedent"
          role="button"
        >
          Précédent
        </a>
      </button>
    </Button>
  </Link>
  <Button
    className="customButtonPagination"
    color="secondary"
    disabled={true}
    tag="button"
  >
    <button
      aria-label={null}
      className="customButtonPagination btn btn-secondary disabled"
      disabled={true}
      onClick={[Function]}
    >
      Page 
      1
       sur 
      <span
        className="totalPages"
      >
        3
      </span>
    </button>
  </Button>
  <Link
    href={
      Object {
        "pathname": "faq",
        "query": Object {
          "page": 2,
        },
      }
    }
  >
    <Button
      className="customActualityButtonPage suivantButton"
      color="secondary"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      tag="button"
    >
      <button
        aria-label={null}
        className="customActualityButtonPage suivantButton btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        onMouseEnter={[Function]}
        type="button"
      >
        <a
          className="suivant"
          role="button"
        >
          Suivant
        </a>
      </button>
    </Button>
  </Link>
</div>
`;
